sys/include/net/ng_ipv6/nc.h

```diff
@@ -0,0 +1,204 @@
+ * more details.
+ */
+
+/**
+ * @defgroup    net_ng_ipv6_nc  IPv6 neighbor cache
+ * @ingroup     net_ng_ipv6
+ * @brief       Translates IPv6 addresses to link layer addresses.
+ * @{
+ *
+ * @file
+ * @brief       Neighbor cache definitions.
+ *
+ * @author      Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+
+#ifndef NG_IPV6_NC_H_
+#define NG_IPV6_NC_H_
+
+#include <stdint.h>
+
+#include "kernel_types.h"
+#include "net/ng_ipv6/addr.h"
+#include "net/ng_netif.h"
+#include "net/ng_pktqueue.h"
+#include "timex.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   The size of the neighbor cache
+ */
+#define NG_IPV6_NC_SIZE             (NG_NETIF_NUMOF * 8)
+
+/**
+ * @{
+ * @name Flag definitions for ng_ipv6_nc_t
+ */
+/**
+ * @{
+ * @brief   States of a neighbor cache entry.
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc4861#section-7.3.2">
+ *          RFC 4861, section 7.3.2
+ *      </a>
+ */
+#define NG_IPV6_NC_STATE_MASK           (0x07)
+#define NG_IPV6_NC_STATE_POS            (0)
+
+#define NG_IPV6_NC_STATE_UNMANAGED      (0x00)  /**< The entry is not manage by NDP */
+
+/**
+ * @brief The entry is unreachable
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc7048#section-3">
+ *          RFC 7048, section 3
+ *      </a>
+ */
+#define NG_IPV6_NC_STATE_UNREACHABLE    (0x01)
+#define NG_IPV6_NC_STATE_INCOMPLETE     (0x02)  /**< Address resolution is performed */
+#define NG_IPV6_NC_STATE_STALE          (0x03)  /**< The entry is stale */
+#define NG_IPV6_NC_STATE_DELAY          (0x04)  /**< The entry was stale but packet was sent out */
+#define NG_IPV6_NC_STATE_PROBE          (0x05)  /**< Periodic reachabality confirmation */
+#define NG_IPV6_NC_STATE_REACHABLE      (0x07)  /**< The entry is reachable */
+/**
+ * @}
+ */
+
+#define NG_IPV6_NC_IS_ROUTER            (0x08)  /**< The neighbor is a router */
+
+#define NG_IPV6_NC_TYPE_MASK            (0x30)
+#define NG_IPV6_NC_TYPE_POS             (4)
+
+/**
+ * @{
+ * @brief   States of a neighbor cache entry.
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc6775#section-3.5">
+ *          RFC 6775, section 3.5
+ *      </a>
+ */
+/**
+ * @brief The entry has no type
+ *
+ * @details The node sents multicast Neighbor Solicitations for hosts.
+ */
+#define NG_IPV6_NC_TYPE_NONE            (0x00)
+#define NG_IPV6_NC_TYPE_GC              (0x10)  /**< The entry is marked for removal */
+#define NG_IPV6_NC_TYPE_TENTATIVE       (0x20)  /**< The entry is temporary */
+#define NG_IPV6_NC_TYPE_REGISTERED      (0x30)  /**< The entry is registered */
+/**
+ * @}
+ */
+/**
+ * @}
+ */
+
+/**
+ * @brief   Neighbor cache entry as defined in
+ *          <a href="http://tools.ietf.org/html/rfc4861#section-3.5#section-5.1">
+ *              RFC 4861, section 5.1
+ *          </a>.
+ */
+typedef struct ng_ipv6_nc {
+    struct ng_ipv6_nc *next;    /**< Next entry for this interface */
+    pktqueue_t pkts;            /**< Packets waiting for address resolution */
```

Comment:
According to the further above referenced [RFC 4861](http://tools.ietf.org/html/rfc4861#section-5.1)

> Entries […] contain such information as […] a pointer to any queued packets waiting for address resolution to complete, etc

And after seeing this I thought this might be the easiest solution to trigger a send of the packet, once the address was resolved. Since you need the entry anyways (to add the address) you can just send all queued packets from here on.